### ISSUE TITLE
Lib is not really crossfading

### ISSUE THREAD

**Issue by maerzhase** [asbjornh/react-tiny-crossfade#4]
Thanks for the lib i used it because i loved the simplicity of the API!
Having to implement a very precise transition i realised that the name is actually a little bit confusing because its never __really__ crossfading. You can set an enter and leave transition but it will always first trigger the leave transition and then use the enter transition for the new component. If one uses this library e.g. for a full size background image you will always have short moment where u see the underlying background color. 

An actual crossfade effect would mean that on 50% both animations are at 50%. Like a DJ ;) when you put the crossfader to 50% you hear two songs at the same time with 50%. Sorry to be the party pooper! 

EDIT:

The difference is also mentioned in one of the libs you mention in the README.

https://github.com/marnusw/react-css-transition-replace#cross-fading-two-components
vs
https://github.com/marnusw/react-css-transition-replace#fade-out-then-fade-in

**Comment by asbjornh** [comment]
I agree. The analogy doesn't really hold for what this component does so the name is misleading.

I have to admit that I've somewhat lost interest in this package so I don't think it's likely that I'll do anything about this in the near future. PR's are welcome of course. The default behaviour should be an actual crossfade as you describe it, but I think the current behavior should be possible to achieve with options.

**Comment by maerzhase** [comment]
No worries on my end. This lib is still helpful in many cases! I ended up using normal `ReactTransitionGroup` to build the animation, was also pretty simple! cheers!
